doc/user_guide/interactive_operations_ROIs.rst

```diff
@@ -0,0 +1,257 @@
+
+
+Interactive Operations and Region of Interest (ROI)
+***************************************************
+
+.. _interactive-label:
+
+Interactive operations
+----------------------
+
+
+The function :py:func:`~.interactive.interactive` ease the task of defining
+operations that are automatically updated when an event is triggered. By
+default it recomputes the operation when data or the axes of the original
+signal changes.
+
+.. code-block:: python
+
+    >>> s = hs.signals.Signal1D(np.arange(10.))
+    >>> ssum = hs.interactive(s.sum, axis=0)
+    >>> ssum.data
+    array([45.0])
+    >>> s.data /= 10
+    >>> s.events.data_changed.trigger(s)
+    >>> ssum.data
+    array([ 4.5])
+
+The interactive operations can be chained.
+
+.. code-block:: python
+
+    >>> s = hs.signals.Signal1D(np.arange(2 * 3 * 4).reshape((2, 3, 4)))
+    >>> ssum = hs.interactive(s.sum, axis=0)
+    >>> ssum_mean = hs.interactive(ssum.mean, axis=0)
+    >>> ssum_mean.data
+    array([ 30.,  33.,  36.,  39.])
+    >>> s.data
+    array([[[ 0,  1,  2,  3],
+            [ 4,  5,  6,  7],
+            [ 8,  9, 10, 11]],
+
+           [[12, 13, 14, 15],
+            [16, 17, 18, 19],
+            [20, 21, 22, 23]]])
+    >>> s.data *= 10
+    >>> s.events.data_changed.trigger(obj=s)
+    >>> ssum_mean.data
+    array([ 300.,  330.,  360.,  390.])
+
+.. _roi-label:
+
+Region Of Interest (ROI)
+------------------------
+
```

Comment:
ROIs can be defined to select part of any compatible signal and may be applied either to the navigation or to the signal axes.